### PR TITLE
Configure compiler and silence warnings of Clang-17

### PR DESCRIPTION
**Pull request by dey4ss**

### CODE REVIEW

**Comment by dey4ss** [comment]
@Bouncner do we want to add a github runner for this with a recent gcc and clang stage?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      fail-fast: false
+      matrix:
+        include:
+          - name: Ubuntu-GCC
+            cc: gcc
+            os: ubuntu-latest
+
+          - name: Ubuntu-Clang
+            cc: clang
+            os: ubuntu-latest
```
Comment (Bouncner) [inline .github/workflows/ci.yml:28]:
```suggestion
            os: ubuntu-latest

          - name: Ubuntu2304-Clang
             cc: clang
             os: ubuntu-23.04
```

Comment (Bouncner) [reply .github/workflows/ci.yml:28]: Soon there will be 24.04 and "our" CI is no longer the latest. Maybe, we should have at least one check for 23.10 (even though is currently the same)?

File: bm_utils.c [hunk 2]
```diff
@@ -575,7 +575,20 @@ set_state(int table, long sf, long procs, long step, long *extra_rows)
 		}
 	*extra_rows = remainder % procs;
 	if (step > procs)	/* moving to the end to generate updates */
-		tdefs[table].gen_seed(*extra_rows);
+        {
+        // HYRISE: The following line raises a compilation error when adding function prototypes the the tdef gen_seed
+        // member, as they epxect two arguments. All seed generators passed in driver.c (L 230-249) take either two
```
Comment (Bouncner) [inline bm_utils.c:580]: expect

File: varsub.c [hunk 3]
```diff
@@ -134,8 +135,9 @@ varsub(int qnum, int vnum, int flags)
 			case 6:
 				tmp_date = UnifInt(93,97,qnum);
                 sprintf(param[1], "19%d-01-01", tmp_date);
-				sprintf(param[2], "0.0%d", UnifInt(2, 9, qnum));
-				sprintf(param[3], "%d", UnifInt((long)24, (long)25, (long)qnum));
+                // HYRISE: Change format specifier.
```
Comment (Bouncner) [inline varsub.c:138]: Alignment looks strange.

**Comment by Bouncner** [comment]
LGTM.

Should we add a line to the README that this fork is using actions and compiles with recent compiler versions?

**Review by Bouncner — APPROVED** [review]

File: varsub.c (hunk 3 above)
Comment (dey4ss) [reply varsub.c:138]: they deliberately mix space and tab alignment. bruh

**Pull request merged**